lighthouse-core/lib/network-request.js

```diff
@@ -0,0 +1,244 @@
+   */
+  onResponseReceived(data) {
+    this._onResponse(data.response, data.timestamp, data.type);
+    this._frameId = data.frameId;
+  }
+
+  /**
+   * @param {LH.Crdp.Network.DataReceivedEvent} data
+   */
+  onDataReceived(data) {
+    this._resourceSize += data.dataLength;
+    if (data.encodedDataLength !== -1) {
+      this.transferSize += data.encodedDataLength;
+      this._transferSize += data.encodedDataLength;
+    }
+  }
+
+  /**
+   * @param {LH.Crdp.Network.LoadingFinishedEvent} data
+   */
+  onLoadingFinished(data) {
+    this.finished = true;
+    this.endTime = data.timestamp;
+    if (data.encodedDataLength >= 0) {
+      this.transferSize = data.encodedDataLength;
+      this._transferSize = data.encodedDataLength;
+    }
+
+    this._updateResponseReceivedTimeIfNecessary();
+  }
+
+  /**
+   * @param {LH.Crdp.Network.LoadingFailedEvent} data
+   */
+  onLoadingFailed(data) {
+    this.finished = true;
+    this.endTime = data.timestamp;
+
+    this.failed = true;
+    this._resourceType = data.type && resourceTypes[data.type];
+    this.localizedFailDescription = data.errorText;
+
+    this._updateResponseReceivedTimeIfNecessary();
+  }
+
+  /**
+   * @param {LH.Crdp.Network.ResourceChangedPriorityEvent} data
+   */
+  onResourceChangedPriority(data) {
+    this.priority = () => data.newPriority;
+  }
+
+  /**
+   * @param {LH.Crdp.Network.RequestWillBeSentEvent} data
+   */
+  onRedirectResponse(data) {
+    if (!data.redirectResponse) throw new Error('Missing redirectResponse data');
+    this._onResponse(data.redirectResponse, data.timestamp, data.type);
+    this.finished = true;
+    this.endTime = data.timestamp;
+
+    this._updateResponseReceivedTimeIfNecessary();
+  }
+
+  /**
+   * @param {LH.Crdp.Network.Response} response
+   * @param {number} timestamp
+   * @param {LH.Crdp.Network.ResponseReceivedEvent['type']=} resourceType
+   */
+  _onResponse(response, timestamp, resourceType) {
+    this.connectionId = String(response.connectionId);
+    this.connectionReused = response.connectionReused;
+
+    if (response.protocol) this.protocol = response.protocol;
+
+    this._responseReceivedTime = timestamp;
+
+    this.transferSize = response.encodedDataLength;
+    this._transferSize = response.encodedDataLength;
+    if (typeof response.fromDiskCache === 'boolean') this._fromDiskCache = response.fromDiskCache;
+
+    this.statusCode = response.status;
+
+    this._timing = response.timing;
+    if (resourceType) this._resourceType = resourceTypes[resourceType];
+    this._mimeType = response.mimeType;
+    this._responseHeaders = NetworkRequest._headersMapToHeadersArray(response.headers);
+
+    this._fetchedViaServiceWorker = !!response.fromServiceWorker;
+
+    if (response.timing) this._recomputeTimesWithResourceTiming(response.timing);
+  }
+
+  /**
+   * @param {LH.Crdp.Network.ResourceTiming} timing
+   */
+  _recomputeTimesWithResourceTiming(timing) {
```

Comment:
add a comment referencing `SDK.NetworkRequest.timing` setter and link to https://github.com/ChromeDevTools/devtools-frontend/blob/56a99365197\/front_end/sdk/NetworkRequest.js#L485-L502

Reply:
done